### PR TITLE
Add support for the airmedia API

### PR DESCRIPTION
**Pull request by foreign-sub**
Add
get_airmedia_configuration
get_airmedia_receivers
send_airmedia
update_airmedia_configuration

### CODE REVIEW

File: aiofreepybox/api/airmedia.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    airmedia_data_schema = {
+        "action": action[0],
+        "media": "",
+        "media_type": media_type[1],
+        "password": "",
+        "position": 0,
+    }
+
+    airmedia_configuration_schema = {"enabled": True}
```
Comment (stilllman) [inline aiofreepybox/api/airmedia.py]:
```suggestion
    airmedia_configuration_schema = {"enabled": True, "password": ""}
```

File: aiofreepybox/api/airmedia.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    async def update_airmedia_configuration(self, airmedia_enabled):
+        """
+        Update airmedia configuration
+
+        airmedia_enabled : `bool`
+        """
+        airmedia_configuration_data = self.airmedia_configuration_schema
+        airmedia_configuration_data["enabled"] = airmedia_enabled
+        return await self._access.put("airmedia/config/", airmedia_configuration_data)
```
Comment (stilllman) [inline aiofreepybox/api/airmedia.py]: Take a `config` object instead of only `enabled`, otherwise the password cannot be changed.

Comment (foreign-sub) [reply aiofreepybox/api/airmedia.py]: i changed it to take parameters instead

File: aiofreepybox/api/airmedia.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    ):
+        """
+        set airmedia configuration
+
+        airmedia_enabled : `bool`
+        airmedia_password : `str`
+        """
+
+        if airmedia_enabled is None and airmedia_password is None:
+            return await self.get_airmedia_configuration()
```
Comment (stilllman) [inline aiofreepybox/api/airmedia.py]:
```suggestion
            return
```

File: aiofreepybox/api/airmedia.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        airmedia_configuration_data = dict
```
Comment (stilllman) [inline aiofreepybox/api/airmedia.py]:
```suggestion
        airmedia_configuration_data = {}
```

File: aiofreepybox/api/airmedia.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        if airmedia_enabled is not None:
+            airmedia_configuration_data["enabled"] = airmedia_enabled
+        if airmedia_password is not None:
+            airmedia_configuration_data["password"] = airmedia_password
+        return await self._access.put("airmedia/config/", airmedia_configuration_data)
```
Comment (stilllman) [inline aiofreepybox/api/airmedia.py]:
```suggestion
        return await self.update_airmedia_configuration(airmedia_configuration_data)
```

File: aiofreepybox/api/airmedia.py (hunk 3 above)
Comment (foreign-sub) [reply aiofreepybox/api/airmedia.py]: fixed.

File: aiofreepybox/api/airmedia.py (hunk 4 above)
Comment (foreign-sub) [reply aiofreepybox/api/airmedia.py]: fixed.

File: aiofreepybox/api/airmedia.py (hunk 5 above)
Comment (foreign-sub) [reply aiofreepybox/api/airmedia.py]: fixed.

**Comment by Quentame** [comment]
Need this for a new [Home Assistant](https://github.com/home-assistant/core) feature @SNoof85

File: aiofreepybox/api/airmedia.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        """
+
+        if airmedia_enabled is None and airmedia_password is None:
+            return None
+        else:
+            airmedia_config: Dict[str, Any] = {}
+            if airmedia_enabled is not None:
+                airmedia_config.update({"enabled": airmedia_enabled})
+            if airmedia_password is not None:
+                airmedia_config.update({"password": airmedia_password})
```
Comment (Quentame) [inline aiofreepybox/api/airmedia.py:80]:
```suggestion

        airmedia_config: Dict[str, Any] = {}
        if airmedia_enabled is not None:
            airmedia_config.update({"enabled": airmedia_enabled})
        if airmedia_password is not None:
            airmedia_config.update({"password": airmedia_password})

```

File: aiofreepybox/api/airmedia.py [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+        if enabled is None:
+            config = await self.get_airmedia_configuration()
+            if config is not None:
+                return config["enabled"]
+        else:
+            config = {"enabled": enabled}
+            apply = await self.set_airmedia_configuration(config)
+            if apply is not None:
+                return apply["enabled"]
+        return None
```
Comment (Quentame) [inline aiofreepybox/api/airmedia.py:100]:
```suggestion
```

`update_airmedia_configuration(airmedia_enabled = True|False)` is doing the job.

**Review by Quentame — CHANGES_REQUESTED** [review]
Improvement would be to use `TypedDict` (for all the lib).
Exemple:
https://github.com/hacf-fr/meteofrance-api/blob/ee08ccb97c8a3e318045aca8a694398038384be0/src/meteofrance/model/forecast.py#L19-L26

**Pull request closed without merging**